bundlr/bundle.go

```diff
@@ -108,3 +110,46 @@ func (b *Bundle) lsDataDirSorted() ([]os.FileInfo, error) {
 func (b *Bundle) ensureDataDir() error {
 	return b.FS().MkdirAll(DataDir, 0700)
 }
+
+func (b *Bundle) Manifest() (*Manifest, error) {
+	if b.manifest != nil {
+		return b.manifest, nil
+	}
+
+	manifestExists, err := afero.Exists(b.fs, manifestFilename);
+	if err != nil {
+		return nil, err
+	}
+	if !manifestExists {
+		return &Manifest{}, nil
+	}
+
+	var manifest Manifest
+	rawManifest, err :=  afero.ReadFile(b.fs, manifestFilename)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(rawManifest, &manifest); err != nil {
+		return nil, err
+	}
+
+	return &manifest, nil
+}
+
+func (b *Bundle) WriteManifest(manifest *Manifest) error {
+	b.rw.Lock()
```

Comment:
This lock is for handling the reading and writing the data content in the `data` subdirectory of a Bundle.

The manifest is its own file, so technically you don't need to obtain this lock to update the Manifest unless you want to call "WriteManifest" concurrently.

But that use case doesn't really make sense to me and it's pretty obvious that the these Manifest related methods are not thread-safe anyway.

So maybe skip the locking.

Reply:
>  it's pretty obvious that the these Manifest related methods are not thread-safe anyway.

Are you saying this because the user can always call `afero.WriteFile(b.FS(), "manifest.json", rawManifest, 600)` externally?
True that can't be prevented, `b.FS()` will grant access to the entire bundle anyway.

Alright then, I'll remove the lock.

Reply:
What I means is the lock was for the reading and writing of `b.DataFS()` basically. The data handling there need to be consistent and coordinated. But the reading and writing of Manifest is totally seperate from that and there will be no conflict between the two.

So `WriteManifest` just need to be not racing against itself on other goroutines. But since even the getter set the `manifest` field on Bundle without any consideration of concurrency, locking here won't make `Manifest` concurrent safe.

And at last, I don't think the Manifest needs to be used concurrently anyway.

Reply:
Agree! I started to add a manifest Lock at the beginning then thought it would need to rename `rw` to `dataLock` to be distinguishable. Eventually, I simplified by re-using `rw`. But ultimately it doesn't make much sense.

Reply:
I know